AndelaOSP/Andela-Socials
# #158187265 Dockerize the application

#### What Does This PR Do?
- Setup docker for the application

#### Description Of Task To Be Completed
- Create `docker-compose.yml` for base application
- Create `Dockerfile` for the server, client and database
- Add setup scripts to the `Dockerfile`

#### Any Background Context You Want To Provide?
- To have a unified environment for the application

#### How can this be manually tested?
- Ensure docker is installed locally on machine
- Ensure these environment variables are added like so `DB_HOST=postgres, DB_NAME=a_socials, DB_USER=a_socials, DB_PASS=a_socials` `CLIENT_PORT=9000`
- To build and spin up the docker containers, run `docker-compose build`
- After building, run the containers with `make docker-build-client` or `make docker-build-server`

#### What are the relevant github issues?
N/A

#### Screenshot(s)

![image](https://user-images.githubusercontent.com/30419014/41624804-a8ad7860-740e-11e8-8b00-5186805fd5c7.png)

![image](https://user-images.githubusercontent.com/30419014/41624627-3bb6decc-740e-11e8-9f4b-95db5fcb0c8b.png)
